### ISSUE TITLE
Rename `sample_weight`?

### ISSUE THREAD

**Issue by wxs** [fchollet/keras#482]
So the `sample_weight` parameter to `fit()` is designed to lets you mask individual timesteps as well. So `sample_weight` is not a great name since it's not actually the weight of a sample. This is causing some confusion in e.g. #451 

Any thoughts on what would be better name?

**Comment by wxs** [comment]
I might go with, e.g. `output_weight`

**Comment by fchollet** [comment]
The name `sample_weight` mimics the equivalent argument in sklearn. I don't think it should be renamed.

The dual behavior can be confusing and should be clarified in the documentation.

**Comment by wxs** [comment]
Fair enough. I've put up a PR #494 for that.
